src/redux/sagas/onboarding.js

```diff
@@ -0,0 +1,51 @@
+import {
+  put, takeLatest, call, select,
+} from 'redux-saga/effects';
+
+import { blockchainActions, onBoardingActions } from '../actions';
+import { blockchainSelectors } from '../selectors';
+import { getComplimentaryLLD } from '../../api/backend';
+import { getBalanceByAddress } from '../../api/nodeRpcCall';
+
+// WORKERS
+
+function* claimComplimentaryLLDWorker() {
+  try {
+    const walletAddress = yield select(blockchainSelectors.userWalletAddressSelector);
+    const getComplimentaryLLDResponse = yield call(getComplimentaryLLD, walletAddress);
+    let errorData = {};
+    if (getComplimentaryLLDResponse.status === 200) {
+      // FIXME eventually we need a notification modal, not just error modal
+      errorData = { details: 'Complimentary LLDs claimed. Please refresh the page' };
```

Comment:
We need a modal to open if clicking a button results in an error, because its important to explain to the user exactly why (and also when they report). Even if we do it via action, a modal will be needed

Reply:
I agree. For errors we do need a modal. But here I'm suggesting ditching the error modal for the success case, when there's no error.

The UX for success case could look like this with no modal:
1. user clicks button to send
2. user sees the loader
3. we do the action
4. we refresh the new isUserEligible... state
5. we close the loader
6. user doesn't have a modal, but sees that loader disappeared, button to claim disappeared and it says that he can't claim anymore because he has already claimed.